File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = 1000
    for idx,target in enumerate(targets):
        count = 0
        for tar in target:
            Min = 999999
            for key in keymap:
                for i,k in enumerate(key):
                    if tar == k and Min > i+1:
                        Min = i+1
                        break
            if Min != 999999:
                count += Min
            else : break
        if Min != 999999:
            targets[idx] = count
        else : targets[idx] = -1
                     
        
    return targets

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    
    
    const answer = targets.map((alpha,a_idx) => {
        let click = 0
        for(let i = 0; i < alpha.length; i++){
            const press = keymap.map((key,k_idx) => {
                for(let j=0; j < key.length; j++){
                    if(key[j] === alpha[i]){return (j+1)}
                }
                return 101
            })
            let Min = Math.min(...press)
            if(Min !== 101){click += Min}
            else {
                click = -1
                break
            }
        }
        return click
    })
    return answer;
}